compiler/aarch64/codegen/OMRMachine.cpp

```diff
@@ -19,11 +19,457 @@
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
  *******************************************************************************/
 
+#include "aarch64/codegen/OMRMachine.hpp"
+
+#include <stdint.h> // for uint16_t, int32_t, etc
+
 #include "codegen/CodeGenerator.hpp"
 #include "codegen/Machine.hpp"
+#include "codegen/Machine_inlines.hpp"
+#include "codegen/RealRegister.hpp" // for RealRegister, etc
+#include "infra/Assert.hpp" // for TR_ASSERT
 
 OMR::ARM64::Machine::Machine(TR::CodeGenerator *cg) :
-      OMR::Machine(cg)
+      OMR::Machine(NUM_ARM64_GPR, NUM_ARM64_FPR)
+   {
+   _registerFile = (TR::RealRegister **)cg->trMemory()->allocateMemory(sizeof(TR::RealRegister *)*TR::RealRegister::NumRegisters, heapAlloc);
+   self()->initialiseRegisterFile();
+   }
+
+TR::RealRegister *OMR::ARM64::Machine::findBestFreeRegister(TR_RegisterKinds rk,
+                                                            bool excludeGPR0,
```

Comment:
Both "p" and "arm" versions have `excludeGPR0`, while the "arm" version does not seem to use it.
It is needed in "p" because `r0` is used as the zero register depending on the instruction?